ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-filter/index.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+
+import { Button, buttonTypeEnum } from '../button'
+
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import * as FilterBuilderView from '../../../component/filter-builder/filter-builder.view'
+import * as FilterBuilderModel from '../../../component/filter-builder/filter-builder'
+
+type Props = {
+  removeFilter: () => void
+  saveFilter: (transformToCql: () => any) => void
+  hasFilter: Boolean
+  resultFilter?: {}
+  isList?: Boolean
+}
+
+const Root = styled.div`
+  display: block;
+  padding: ${props => props.theme.minimumSpacing};
+`
+
+const EditorFooter = styled.div`
+  padding-top: 30px;
+  white-space: nowrap;
+`
+
+const SaveButtonStyle = (props: Props) =>
+  ({
+    display: 'inline-block',
+    width: (props.hasFilter && '50%') || '100%',
+  } as React.CSSProperties)
+
+const RemoveButtonStyle = {
+  display: 'inline-block',
+  width: '49%',
+  marginRight: '1%',
+} as React.CSSProperties
+
+class View extends React.Component<Props, { filterCqlCallback: () => any }> {
+  view: any
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      filterCqlCallback: () => {},
+    }
```

Comment:
```suggestion
        this.view = new FilterBuilderView({
            model: new FilterBuilderMOdel({isResultFilter: true})
        });
    }
```

As I mentioned in the above comment in marionette-region-container, this will avoid the need for `bindView`.

Reply:
@andrewkfiedler I agree, this will work for views that have no cascading child regions/views that they 'mount', but `filter-builder.view` calls `turnOnEditing` on several regions itself: https://github.com/codice/ddf/blob/f1d6bf5a159b4cd1ece220c03e843c1c17b251d1/ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.view.js#L182 So what I'm seeing is that even though `marionette-region-container` will 'show' `FuilterBuilderView` as soon as I attempt to call `turnOnEditing` on it (from within `componentDidMount`) I get an exception storm for those regions being null

Reply:
Since we know that `onBeforeShow` is the moment that it's safe to do what we need, we can create a slightly modified version of the FilterBuilderView that will do exactly what we want.  
```
const CustomFilterBuilderView = FilterBuilderView.extend({
  onBeforeShow() {
    FilterBuilderView.prototype.onBeforeShow.call(this)
    this.turnOnEditing()
    this.turnOffNesting()
  },
})
```
This keeps the logic located as close as possible to what it depends on and avoids any uncertainty as to whether or not child views will be ready.  When onBeforeShow is called, the view will be constructed and then editing and nesting will be turned off.